# **AVIATION DATASET DATA ANALYSIS PROJECT**

## **1. Objectives**<hr>

#### **Business Question**<br>
Your company is expanding in to new industries to diversify its portfolio. Specifically, they are interested in purchasing and operating airplanes for commercial and private enterprises, but do not know anything about the potential risks of aircraft. You are charged with determining which aircraft are the lowest risk for the company to start this new business endeavor. You must then translate your findings into actionable insights that the head of the new aviation division can use to help decide which aircraft to purchase.

### Objective 1<br>
Which aircraft is the safest?

### Objective 2<br>
Identify the safest and most dangerous countries & locations to fly over

### Objective 3<br>
The most common type of accidents so as to invest in advance in prevetive measures

<br>

## **2. Data Understanding**

In this section we try to understand the data itself. First we have to read the source notes on what is in the columns, the categories in categorical and what missing data means. We look at its shape and also at the descriptive statistics of the data.

**Importing libraries and data**

In [70]:
#Importing libraries
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [71]:
#Importing the data
acc_df = pd.read_csv("data/aviation-accident-data-2023-05-16.csv")

Lets look at the top and bottom 10 rows of the dataset

In [72]:
acc_df.head(10)

date             type registration  \
0  date unk.   Antonov An-12B       T-1206   
1  date unk.   Antonov An-12B       T-1204   
2  date unk.   Antonov An-12B       T-1201   
3  date unk.  Antonov An-12BK          NaN   
4  date unk.  Antonov An-12BP   CCCP-11815   
5  date unk.  Antonov An-12BP   CCCP-12172   
6  date unk.     Antonov An-2    CCCP-N574   
7  date unk.     Antonov An-2   CCCP-01216   
8  date unk.   Antonov An-24B     RA-47794   
9  date unk.    Antonov An-26       01 red   

                               operator fatalities             location  \
0                         Indonesian AF        NaN                  NaN   
1                         Indonesian AF        NaN                  NaN   
2                         Indonesian AF        NaN                  NaN   
3                             Soviet AF        NaN  Tiksi Airport (IKS)   
4                             Soviet AF          0  Massawa Airport ...   
5                             Soviet AF        NaN                  NaN   
6  GUSMP, Directorate of Polar Aviation        NaN              unknown   
7                              Aeroflot          0         Chita region   
8                            Russian AF          0                  NaN   
9                             Soviet AF          0    Orenburg Air Base   

           country cat     year  
0  Unknown country  U1  unknown  
1  Unknown country  U1  unknown  
2  Unknown country  U1  unknown  
3           Russia  A1  unknown  
4          Eritrea  A1  unknown  
5           Russia  U1  unknown  
6           Russia  A1  unknown  
7           Russia  A2  unknown  
8           Russia  A1  unknown  
9           Russia  O1  unknown

In [73]:
acc_df.tail(10)

date                       type registration  \
23957  21-APR-2023           Ilyushin Il-76TD       ST-BDN   
23958  21-APR-2023   Beech 200 Super King Air       LV-WPM   
23959  26-APR-2023           Boeing 777-233LR       C-FIUF   
23960  09-MAY-2023     Cessna 208 Caravan 675       PK-HVG   
23961  10-MAY-2023                Learjet 36A        N56PA   
23962  11-MAY-2023               Hawker 900XP       PK-LRU   
23963  11-MAY-2023  Cessna 208B Grand Caravan       PK-NGA   
23964  12-MAY-2023  Cessna 208B Grand Caravan       5X-RBR   
23965  14-MAY-2023            Boeing 747-4R7F       LX-OCV   
23966  15-MAY-2023                Learjet 35A       D-CGFQ   

                     operator fatalities                  location  \
23957           Badr Airlines          0       Khartoum Interna...   
23958               Flytec SA          0       Salta-Martín Mi...   
23959              Air Canada          0       Sydney-Kingsford...   
23960             Dimonim Air          0             Yabi Airstrip   
23961             Phoenix Air          3  near San Clemente Isl...   
23962  Angkasa Super Services          0       Maleo Airport (MOH)   
23963  Nasional Global Aviasi          0         Fentheik Airstrip   
23964            Bar Aviation          0       Kampala-Kajjansi...   
23965                Cargolux          0       Luxembourg-Finde...   
23966                     GFD          2             Hohn Air Base   

          country cat  year  
23957       Sudan  O1  2023  
23958   Argentina  A2  2023  
23959   Australia  A2  2023  
23960   Indonesia  A2  2023  
23961         USA  A1  2023  
23962   Indonesia  A2  2023  
23963   Indonesia  A2  2023  
23964      Uganda  A1  2023  
23965  Luxembourg  A2  2023  
23966     Germany  A1  2023

At a first glance we get a rough Idea about the format of data in cells. We can also see what null valuesare denoted as especialy in the non-number columns, where they are denoted as unknown

In [74]:
# Getting the shape of data
acc_df.shape

(23967, 9)

The dataframe has 23967 rows and 9 columns. Any column with less than 23967 entries has null values

In [75]:
# Lets look at the column infomation
acc_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 23967 entries, 0 to 23966
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   date          23967 non-null  object
 1   type          23967 non-null  object
 2   registration  22419 non-null  object
 3   operator      23963 non-null  object
 4   fatalities    20029 non-null  object
 5   location      23019 non-null  object
 6   country       23967 non-null  object
 7   cat           23967 non-null  object
 8   year          23967 non-null  object
dtypes: object(9)
memory usage: 1.6+ MB


In [76]:
acc_df["fatalities"].describe()

count     20029
unique      369
top           0
freq      10713
Name: fatalities, dtype: object

We can see all the columns are of object datatype. Fatalities column off the rip needs to be converted to a number. Registration, operator, fatalities and location columns all have null values that have to be dealt with.

<br>

**Descriptive Statistics of the data**

In this section we get the descriptive statistics of the columns

In [77]:
acc_df.describe()

date                  type registration operator fatalities  \
count         23967                 23967        22419    23963      20029   
unique        15079                  3201        21962     6017        369   
top     10-MAY-1940  Douglas C-47A (DC-3)       LZ-...    USAAF          0   
freq            171                  1916           13     2604      10713   

       location country    cat   year  
count     23019   23967  23967  23967  
unique    14608     232     11    106  
top     unknown     USA     A1   1944  
freq        272    4377  17424   1505

Here we can see the descriptive statistics of the non numerical data. We can infer:

    - A1 is the most common accident.
    - 10th May 1940 had alot of accidents
    - Most accidents occured in the year 1944
    - Most accidents had no fatalities.
    - Most accidents occured in the US.
    - The countries might have spelling errors since there are 232 unique countries compared to the 195-200 countries in the world.

In [ ]:
# Converting the fatalities  column to int datatype to get correct descriptive statistics
acc_df["fatalities"] = pd.to_numeric(acc_df["fatalities"], errors="coerce") 
#Converts string values to int and if the srting is in a non-int format it isrplaced with NaN

In [79]:
acc_df["fatalities"].describe()

count    19676.000000
mean         6.135851
std         18.686195
min          0.000000
25%          0.000000
50%          0.000000
75%          5.000000
max        520.000000
Name: fatalities, dtype: float64

We can see most accidents were non fatal. Also mean is not a good measure of central tendency here since the data is left-skewed.